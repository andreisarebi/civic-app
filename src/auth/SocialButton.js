import React from 'react';
import { Image, StyleSheet, View } from 'react-native';
import { Button } from 'react-native-elements';
import PropTypes from 'prop-types';
import colors from '../styles/colors';

const GoogleIcon = props => (
  <View style={styles.iconContainer}>
    <Image
      {...props}
      source={require('../assets/images/google-icon.png')}
      style={styles.googleIcon}
    />
  </View>
);

const FacebookIcon = props => (
  <View style={styles.iconContainer}>
    <Image
      {...props}
      source={require('../assets/images/facebook-icon.png')}
      style={styles.facebookIcon}
    />
  </View>
);

const SocialButton = props => {
  const config = (type => {
    switch (type) {
      case 'email':
        return {
          textColor: colors.black,
          icon: 'email',
        };
      case 'facebook':
        return {
          textColor: colors.white,
          icon: FacebookIcon,
        };
      case 'google':
        return {
          textColor: colors.black,
          icon: GoogleIcon,
        };
    }
  })(props.type.toLowerCase());

  return (
    <Button
      borderRadius={3}
      buttonStyle={[styles.button, styles[props.type]]}
      color={config.textColor}
      containerViewStyle={props.style}
      fontSize={18}
      icon={{ name: config.icon }}
      iconComponent={config.icon}
      title={props.title}
      onPress={props.onPress}
      outline={true}
      textStyle={styles.textStyle}
    />
  );
};

SocialButton.propTypes = {
  onPress: PropTypes.func,
  style: PropTypes.object,
  title: PropTypes.string,
  type: PropTypes.oneOf(['google', 'facebook', 'email']),
};

const styles = StyleSheet.create({
  button: {
    alignSelf: 'stretch',
  },
  // eslint-disable-next-line
  email: {
    backgroundColor: colors.white,
    borderColor: colors.lightGray2,
  },
  // eslint-disable-next-line
  facebook: {
    backgroundColor: colors.facebookBlue,
    borderColor: colors.facebookBlue,
  },
  facebookIcon: {
    height: 35,
    width: 35,
  },
  // eslint-disable-next-line
  google: {
    backgroundColor: colors.white,
    borderColor: colors.lightGray2,
  },
  googleIcon: {
    height: 32,
    width: 32,
  },
  iconContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  textStyle: {
    flex: 3,
  },
});

export default SocialButton;
