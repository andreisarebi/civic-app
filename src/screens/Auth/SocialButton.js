import React from 'react';
import { StyleSheet } from 'react-native';
import { Button } from 'react-native-elements';
import PropTypes from 'prop-types';
import colors from '../../styles/colors';

const SocialButton = props => (
  <Button
    borderRadius={3}
    buttonStyle={[styles.button, props.buttonStyle]}
    color={props.textColor}
    containerViewStyle={props.style}
    fontSize={18}
    icon={props.icon}
    iconComponent={props.iconComponent}
    title={props.title}
    onPress={props.onPress}
    outline
    textStyle={styles.textStyle}
  />
);

SocialButton.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  icon: PropTypes.shape({
    name: PropTypes.string,
    type: PropTypes.string,
    color: PropTypes.string,
    size: PropTypes.number,
  }),
  textColor: PropTypes.string,
  iconComponent: PropTypes.func,
  style: Button.propTypes.containerViewStyle,
  buttonStyle: Button.propTypes.buttonStyle,
};

SocialButton.defaultProps = {
  textColor: colors.black,
  icon: {
    name: 'email',
    type: 'material-community',
    color: colors.black,
    size: 35,
  },
  iconComponent: null,
  style: {},
  buttonStyle: {},
};

const styles = StyleSheet.create({
  button: {
    alignSelf: 'stretch',
    backgroundColor: colors.white,
    borderColor: colors.lightGray2,
  },
  textStyle: { flex: 3 },
});

export default SocialButton;
