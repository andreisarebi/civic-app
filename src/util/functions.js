import distanceInWordsStrict from 'date-fns/distance_in_words_strict';

export const distanceInWordsFromNow = (date) => {
  return distanceInWordsStrict(date, Date.now());
}
