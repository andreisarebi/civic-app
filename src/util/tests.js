import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import store from '../../App/configureStore';

export const createSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Component {...props} />);
  return renderer.getRenderOutput();
};

export const createConnectedSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Provider store={store}><Component {...props} /></Provider>);
  return renderer.getRenderOutput();
};
