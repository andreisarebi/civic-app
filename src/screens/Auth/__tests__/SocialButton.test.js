import { createSnapshot } from '../../../util/tests';
import SocialButton from '../SocialButton';

describe('SocialButton', () => {
  it('renders correctly', () => {
    const snapshot = createSnapshot(SocialButton, { title: 'Sign In with Email', onPress: jest.fn() });
    expect(snapshot).toMatchSnapshot();
  });
});
