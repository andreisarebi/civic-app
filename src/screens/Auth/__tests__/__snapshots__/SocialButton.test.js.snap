// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialButton email renders correctly 1`] = `
<Button
  borderRadius={3}
  buttonStyle={
    Array [
      Object {
        "alignSelf": "stretch",
      },
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#e0e0e0",
      },
    ]
  }
  color="#000000"
  containerViewStyle={undefined}
  fontSize={18}
  icon={
    Object {
      "color": "#000000",
      "name": "email",
      "size": 35,
      "type": "material-community",
    }
  }
  iconComponent={undefined}
  onPress={undefined}
  outline={true}
  textStyle={
    Object {
      "flex": 3,
    }
  }
  title="Sign In with Email"
/>
`;

exports[`SocialButton facebook renders correctly 1`] = `
<Button
  borderRadius={3}
  buttonStyle={
    Array [
      Object {
        "alignSelf": "stretch",
      },
      Object {
        "backgroundColor": "#3c5a9a",
        "borderColor": "#3c5a9a",
      },
    ]
  }
  color="#ffffff"
  containerViewStyle={undefined}
  fontSize={18}
  icon={
    Object {
      "name": "facebook-box",
      "size": 35,
      "type": "material-community",
    }
  }
  iconComponent={undefined}
  onPress={undefined}
  outline={true}
  textStyle={
    Object {
      "flex": 3,
    }
  }
  title="Continue with Facebook"
/>
`;

exports[`SocialButton google renders correctly 1`] = `
<Button
  borderRadius={3}
  buttonStyle={
    Array [
      Object {
        "alignSelf": "stretch",
      },
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#e0e0e0",
      },
    ]
  }
  color="#000000"
  containerViewStyle={undefined}
  fontSize={18}
  icon={Object {}}
  iconComponent={[Function]}
  onPress={undefined}
  outline={true}
  textStyle={
    Object {
      "flex": 3,
    }
  }
  title="Sign In with Google"
/>
`;
